src/platform/packages/private/kbn-ui-shared-deps-npm/webpack.config.js

```diff
@@ -138,7 +140,19 @@ module.exports = (_, argv) => {
 
     resolve: {
       alias: {
-        '@elastic/eui$': '@elastic/eui/optimize/es',
+        // @elastic/eui-amsterdam is a package alias defined in Kibana's package.json
+        // that points to special EUI releases bundled with Amsterdam set as the default theme
+        // and meant to be used with Kibana 8.x. Kibana 9.0 and later use the Borealis theme
+        // and should import from the regular @elastic/eui package.
+        '@elastic/eui$': useEuiAmsterdamRelease
+          ? '@elastic/eui-amsterdam/optimize/es'
+          : '@elastic/eui/optimize/es',
+        '@elastic/eui/optimize/es/components/provider/nested$': useEuiAmsterdamRelease
```

Comment:
I can be wrong but last time I checked I thought we are not using the `@elastic/eui/optimize/es/components|services` here but I don't think it is hurting having those here as well